collectors/proc.plugin/proc_meminfo.c

```diff
@@ -219,7 +219,8 @@ int do_proc_meminfo(int update_every, usec_t dt) {
 
     unsigned long long SwapUsed = SwapTotal - SwapFree;
 
-    if(do_swap == CONFIG_BOOLEAN_YES || SwapTotal || SwapUsed || SwapFree) {
+    if(do_swap == CONFIG_BOOLEAN_YES || (do_swap == CONFIG_BOOLEAN_AUTO &&
+                                         (SwapTotal || SwapUsed || SwapFree || netdata_zero_metrics_enabled == CONFIG_BOOLEAN_YES))) {
         do_swap = CONFIG_BOOLEAN_YES;
```

Comment:
Could you format it better? It looks ugly and it is hard to read, it is even longer then 120 characters. Do you like it yourself?

![Screenshot_20190710_114014](https://user-images.githubusercontent.com/22274335/60954400-94ea6a80-a307-11e9-9afa-067f9af456d5.png)

Reply:
This is true for all conditions. I believe there is always a possibility to make them readable.

Reply:
A graceful solution needs a great deal of refactoring. So I won't do it for now.